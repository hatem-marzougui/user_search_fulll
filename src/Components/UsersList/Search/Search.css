.search-container {
  width: 100%;
  height: 10vh;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid blue;
}
.search-input {
  text-align: center;
  width: 100%;
  max-width: 500px;
  padding: 10px 15px;
  font-size: 1rem;
  border: 1px solid #ccc;
  border-radius: 8px;
  transition: border-color 0.3s ease;
}
