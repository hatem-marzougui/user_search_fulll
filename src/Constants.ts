export const HEADER_TITLE: string = "Github Search";
export const VIEW_PROFILE_BUTTON_TEXT: string = "View Profile";
export const LOGIN: string = "Login";
export const EDIT_MODE: string = "Edit Mode";
export const DUPLICATE_USERS: string = "Duplicate Users";
export const SEARCH_PLACEHOLDER: string = "Search...";
export const USERS_API_ENDPOINT: string = "/search/users";
export const RATE_LIMIT_ERROR_MESSAGE: string =
  "GitHub request limit reached. Please try again in a minute.";
export const EMPTY_SEARCH_QUERY_ERROR: string = "Please enter a search query";
export const NO_RESULTS_FOUND: string = "No result found  ";
export const USER_CARD_AVATAR_ALT: string = "User Avatar";
export const USER_CARD_ID_LABEL: string = "ID:";
export const USER_CARD_LOGIN_LABEL: string = "Login:";
export const USER_CARD_VIEW_PROFILE_BUTTON_TEXT: string = "View Profile";
export const RATE_LIMIT: string = "rate limit";
export const ELEMENT_SELECTED: string = "elements selected";
