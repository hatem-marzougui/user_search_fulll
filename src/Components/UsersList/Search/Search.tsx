import React from "react";
import "./Search.css";
import { SEARCH_PLACEHOLDER } from "../../../Constants";

type SearchProps = {
  onSearchChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  value: string;
};

const Search = ({ onSearchChange, value }: SearchProps) => {
  return (
    <div className="search-container">
      <input
        type="search"
        value={value}
        onChange={onSearchChange}
        placeholder={SEARCH_PLACEHOLDER}
        className="search-input"
      />
    </div>
  );
};

export default Search;
