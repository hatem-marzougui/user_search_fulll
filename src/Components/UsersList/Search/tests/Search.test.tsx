import { render, screen, fireEvent } from "@testing-library/react";
import Search from "../Search";
import { SEARCH_PLACEHOLDER } from "../../../../Constants";

describe("Search component", () => {
  test("renders input with placeholder", () => {
    render(<Search value="" onSearchChange={() => {}} />);
    const input = screen.getByPlaceholderText(SEARCH_PLACEHOLDER);
    expect(input).toBeInTheDocument();
  });

  test("calls onSearchChange on typing", () => {
    const handleSearchChange = jest.fn();
    render(<Search value="" onSearchChange={handleSearchChange} />);
    const input = screen.getByPlaceholderText(SEARCH_PLACEHOLDER);
    fireEvent.change(input, { target: { value: "test" } });
    expect(handleSearchChange).toHaveBeenCalledTimes(1);
  });

  test("matches snapshot", () => {
    const { container } = render(<Search value="" onSearchChange={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
