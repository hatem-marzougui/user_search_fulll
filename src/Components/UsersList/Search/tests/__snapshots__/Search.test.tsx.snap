// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches snapshot 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      class="search-input"
      placeholder="Rechercher..."
      type="search"
      value=""
    />
  </div>
</div>
`;
